app/controllers/concerns/basic_authenticable.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Enforces HTTP Basic Auth
+module BasicAuthenticable
+  class << self
+    def required?
+      Settings.basic_auth_required
+    end
+
+    def authenticate(username, password)
+      validate(username, password)
+    end
+
+    def validate(username, password)
+      utils.secure_compare(::Digest::SHA256.hexdigest(auth_username), ::Digest::SHA256.hexdigest(username)) &
+        utils.secure_compare(::Digest::SHA256.hexdigest(auth_password), ::Digest::SHA256.hexdigest(password))
+    end
+
+    def auth_username
+      @auth_username ||= Settings.basic_auth.username
+    end
+
+    def auth_password
+      @auth_password ||= Settings.basic_auth.password
```

Comment:
Can have add a little bit to Readme for my gone=>:banana:  password

Reply:
I have no idea what you mean 😄

Reply:
Discussed offline, I know what he means

Reply:
done in https://github.com/DFE-Digital/register-trainee-teacher-data/pull/32/commits/050540f92679ec0e50df0374d509b640a3054731